src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,71 @@
+package duke.task;
+
+/**
+ * Represent task to be done.
+ * Abstract class and cannot be instantiated
+ */
+public abstract class Task {
+    private final String taskName;
+    private boolean isDone;
+
+    public Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+    }
+
+    public Task(String taskName, boolean isDone) {
+        this.taskName = taskName;
+        this.isDone = isDone;
+    }
+
+
+    /**
+     * Check whether a keyword exists in the taskname
+     *
+     * @param word keyword search
+     * @return keyword exists inside the string taskname
+     */
+    public boolean contains(String word) {
```

Comment:
Boolean methods should be named to sound like booleans, so maybe we could modify it to `hasWord` to make it more readable :)